src/overlays/actors/ovl_Obj_Wind/z_obj_wind.c

```diff
@@ -28,21 +29,144 @@ ActorInit Obj_Wind_InitVars = {
-static InitChainEntry D_80B24480[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 4000, ICHAIN_STOP),
 };
 
-#endif
+void ObjWind_Init(Actor* thisx, PlayState* play) {
+    ObjWind* this = (ObjWind*)thisx;
+    s32 pad;
+    WaterBox* waterbox;
+    f32 ySurface;
 
-extern InitChainEntry D_80B24480[];
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if (WaterBox_GetSurface1(play, &play->colCtx, this->actor.world.pos.x, this->actor.world.pos.z, &ySurface,
+                             &waterbox) &&
+        (this->actor.world.pos.y < ySurface)) {
+        this->underWater = 1;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wind/ObjWind_Init.s")
+void ObjWind_Destroy(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wind/ObjWind_Destroy.s")
+typedef struct sD_80B2448C {
+    s16 unk_0;
+    s16 unk_2;
+    s16 unk_4;
+    s16 unk_6;
+    s16 unk_8;
+} sD_80B2448C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wind/ObjWind_Update.s")
+static sD_80B2448C D_80B2448C[] = {
+    { 0x0212, 0x005A, 0x0BB8, 0x07D0, 0x0000 }, { 0x0460, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x026C, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0654, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0258, 0x0078, 0x0BB8, 0x07D0, 0x0000 }, { 0x01F4, 0x0050, 0x0BB8, 0x07D0, 0x0000 },
+    { 0x0190, 0x003C, 0x07D0, 0x07D0, 0x0000 }, { 0x01D6, 0x0046, 0x07D0, 0x03E8, 0x0000 },
+};
+
+static s32 D_80B245CC = 0xFFFFFFFF;
+
+void ObjWind_Update(Actor* thisx, PlayState* play) {
+    ObjWind* this = (ObjWind*)thisx;
+    sD_80B2448C* entry = &D_80B2448C[(this->actor.params >> 7) & 0x1F];
+    Player* player;
+    f32 cosYTimesX; // sp80
+    f32 cosX;       // sp7C
```

Comment:
```suggestion
    f32 cosYTimesX;
    f32 cosX;
```
Not needed anymore (applies to the rest of them)